src/sage/rings/polynomial/polynomial_ring.py

```diff
@@ -1448,24 +1466,64 @@ def random_element(self, degree=(-1,2), *args, **kwds):
             else:
                 raise ValueError("No polynomial of degree >= 0 has all coefficients zero")
 
-        # Pick a random degree
-        d = randint(degree[0], degree[1])
+        if degree == (-1, -1):
+            return 0
```

Comment:
```suggestion
        if degree == (-1, -1):
            return self.zero()
```
The Python `int` `0` is not the same as the zero in the polynomial ring; it has different methods.